packages/access-client/src/agent.js

```diff
@@ -128,67 +126,240 @@ export class Agent {
   }
 
   /**
-   * @param {string} email
-   * @param {object} [opts]
-   * @param {AbortSignal} [opts.signal]
+   *
+   * @param {Ucanto.Delegation} delegation
    */
-  async createAccount(email, opts) {
-    const account = await this.store.createAccount()
-    const service = await this.service()
-    const delegationToAgent = await Any.delegate({
-      issuer: account,
+  async addProof(delegation) {
+    validate(delegation, {
+      checkAudience: this.issuer,
+      checkIsExpired: true,
+    })
+
+    this.data.delegations.set(delegation.cid.toString(), {
+      delegation,
+    })
+
+    await this.store.save(this.data)
+  }
+
+  /**
+   * @param {import('@ucanto/interface').Capability[]} [caps]
+   */
+  async *#delegations(caps) {
+    const _caps = new Set(caps)
+    for (const [key, value] of this.data.delegations) {
+      // check expiration
+      if (!isExpired(value.delegation)) {
+        // check if delegation can be used
+        if (!isTooEarly(value.delegation)) {
+          // check if we need to filter for caps
+          if (caps) {
+            for (const cap of _caps) {
+              if (canDelegateCapability(value.delegation, cap)) {
+                _caps.delete(cap)
+                yield value
+              }
+            }
+          } else {
+            yield value
+          }
+        }
+      } else {
+        // delete any expired delegation
+        this.data.delegations.delete(key)
+      }
+    }
+
+    await this.store.save(this.data)
+  }
+
+  /**
+   * @param {import('@ucanto/interface').Capability[]} [caps]
+   */
+  async *proofs(caps) {
+    for await (const value of this.proofsWithMeta(caps)) {
```

Comment:
Can they please be collected for me? I'm not sure when I would not just collect them all before sending them...